docker/php/docker-entrypoint.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+set -e
+
+# first arg is `-f` or `--some-option`
+if [ "${1#-}" != "$1" ]; then
+	set -- php-fpm "$@"
+fi
+
+if [ "$1" = 'php-fpm' ] || [ "$1" = 'php' ] || [ "$1" = 'bin/console' ]; then
+	# todo: a vérifier dans le container et l'image.
+	PHP_INI_RECOMMENDED="$PHP_INI_DIR/bolt.prod.ini"
+	if [ "$APP_ENV" != 'prod' ]; then
+		PHP_INI_RECOMMENDED="$PHP_INI_DIR/bolt.dev.ini"
+	fi
+	ln -sf "$PHP_INI_RECOMMENDED" "$PHP_INI_DIR/php.ini"
+
+	mkdir -p public/files public/theme public/thumbs var/cache var/log
+	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX config public/files public/theme public/thumbs var
+	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX config public/files public/theme public/thumbs var
+
+	if [ "$APP_ENV" != 'prod' ]; then
+		composer install --prefer-dist --no-progress --no-suggest --no-interaction
+	fi
+
+	echo "Waiting for db to be ready..."
+	ATTEMPTS_LEFT_TO_REACH_DATABASE=60
+	until [ $ATTEMPTS_LEFT_TO_REACH_DATABASE -eq 0 ] || bin/console doctrine:query:sql "SELECT 1" > /dev/null 2>&1; do
+		sleep 1
+		ATTEMPTS_LEFT_TO_REACH_DATABASE=$((ATTEMPTS_LEFT_TO_REACH_DATABASE-1))
+		echo "Still waiting for db to be ready... Or maybe the db is not reachable. $ATTEMPTS_LEFT_TO_REACH_DATABASE attempts left"
+	done
+
+	if [ $ATTEMPTS_LEFT_TO_REACH_DATABASE -eq 0 ]; then
+		echo "The db is not up or not reachable"
+		exit 1
+	else
+		echo "The db is now ready and reachable"
+		php bin/console bolt:info
+	fi
+
+	if ! ls -A public/theme/* > /dev/null 2>&1; then
+		composer run post-create-project-cmd
+	fi
+
+	if ls -A migrations/*.php > /dev/null 2>&1; then
```

Comment:
This part I do not like it too much but I have not found better. I prefer to explain my reasoning to you.

For the construction of the database I am in favor of [doctrine migrations](https://symfony.com/doc/master/bundles/DoctrineMigrationsBundle/index.html) for all environments and especially for tests and production.
```
	if ls -A migrations/*.php > /dev/null 2>&1; then
		bin/console doctrine:migrations:migrate --no-interaction
	fi
```
Do you want me to add depending on the project?
I can add it in the documentation, but where?

Migrations are cool but when we start the project we don't have any yet.
To overcome this problem and only on dev and test environments, I implemented this:
1. Reset the database with the envrioment variable `RESET_DATABASE`
2. Launch of bolt fixtures with the envrioment variable `LAUNCH_FIXTURES`
3. The create admin user with envrioment variables `ADMIN_USERNAME`, `ADMIN_PASSWORD` & `ADMIN_EMAIL`
```
	if [ "$APP_ENV" != 'prod' ]; then
		if [ ! -z $RESET_DATABASE ]; then
			bin/console doctrine:schema:drop --force --full-database
			bin/console doctrine:schema:create
		fi

		if [ ! -z $LAUNCH_FIXTURES ]; then
			bin/console doctrine:fixtures:load --no-interaction
		fi

		if [ ! -z $ADMIN_USERNAME ] && [ ! -z $ADMIN_PASSWORD ] && [ ! -z $ADMIN_EMAIL ]; then
			bin/console bolt:add-user $ADMIN_USERNAME $ADMIN_PASSWORD $ADMIN_EMAIL $ADMIN_USERNAME --admin --no-interaction
		fi
	fi
```

This is documented in the Readme file in the Docker part when creating the `docker-entrypoint.override.yml` file.
This forces the person to read the documentation carefully otherwise the project could potentially not work if it does not use one of the 2 solutions for the creation of the database (create `doctrine migrations` or add `RESET_DATABASE` at `true` in the `docker-entrypoint.override.yml` file)

What do you think @bobdenotter, @alanpoulain ?

Reply:
Personally, I wouldn't like the database to be migrated / reset each time I start the containers.
I think this entrypoint shouldn't do anything if there is no `docker-entrypoint.override.yml`.
The migrations should also be applied only if the user has explicitly tells so IMO.

Reply:
What I did with the different conditions set up. I leave the choice to the user to use it or not ;)

Reply:
> Personally, I wouldn't like the database to be migrated / reset each time I start the containers.
> I think this entrypoint shouldn't do anything if there is no `docker-entrypoint.override.yml`.
> The migrations should also be applied only if the user has explicitly tells so IMO.

Same here. I'm using docker a lot and prefer the start-up of the container(s) to not do anything that isn't necessary. If the default is 'do nothing' I'd be completely ok with extra features, but watch out for maintenance!

Reply:
@simongroenewolt

What I wanted to do is to have a Bolt project that works as soon as we start it without additional commands.
We need a viable database for Bolt to be functional at startup without further intervention. That's why I proposed this system there.

So I have proposed several solution that starts when we tell them explicitly.

If the Doctrine miagrations folder is present, then it builds the database with that.
If we have nothing, I leave the possibility to the developer:
- either build the database with the env variable `$RESET_DATABASE`
- either inject the Bolt fixtures into the database with the env variable `$LAUNCH_FIXTURES`
- either add a new administrator in the database with the env variables `$ADMIN_USERNAME`, `$ADMIN_PASSWORD` & `$ADMIN_EMAIL`

I grant you, it's very practical when testing Bolt and having to start from scratch or when you initialize the project the first time. For the long term of course the developer will do what he wants.

After that, this is only a proposal to prevent the developer from placing additional orders.
I can remove all of that and document the commands to run directly with Docker if you mind?

Reply:
@toofff I understand your goal of wanting to get the developer up and running with minimal interaction. I know that bolt/core had a migration in the repository once, but Bob told me that didn't belong there as the migrations are db specific (mysql/sqlite/postgres).
So I'm not really expecting bolt/platform to supply migrations in the package download either. If the only migrations that are expected will have been generated by the developer they don't need to be auto-run at start do they? Or am I missing something?

Reply:
If the user chooses to use doctrine migrations, then they must be launched from the container run, for all environments, otherwise the project will launch and will fall into error because the database will not be up to date depending on the entities set up.

The best practice when using doctrine migrations is to create the test database with it instead of doing a `doctrine:schema:update --force` or `bin/console doctrine:schema:create`. So we indirectly test our doctrines migrations by launching our various tests. :smile:

Reply:
Since this part was a bit controversial (even for me), I went back to it and removed the addition of a `docker-compose.override.yml` file. Obviously the commands launched automatically are replaced by good documentation in the `README.md` file.